pandera/typing/common.py

```diff
@@ -40,44 +40,93 @@
 #: ``"string"`` pandas dtypes: pandas 1.0.0+. For <1.0.0, this enum will
 #: fall back on the str-as-object-array representation.
 STRING = pandas_engine.STRING  #: ``"str"`` numpy dtype
+Geometry = pandas_engine.Geometry  #: ``"geometry"`` geopandas dtype
+
+try:
+    import geopandas as gpd
+
+    GEOPANDAS_INSTALLED = True
+except ImportError:
+    GEOPANDAS_INSTALLED = False
+
+if GEOPANDAS_INSTALLED:
```

Comment:
I think we can just add all possible types to this `TypeVar` so that we don't need to do this conditional?

We need to think about the typing implications here @jeffzi your thoughts would be much-appreciated here!

Reply:
The `Geometry` type will only be available when geopandas is installed, so that's why I ended up doing this

Reply:
ah, right. would something like this work?

```
generic_types = [
    bool,
    int,
    ...
]

if GEOPANDAS_INSTALLED:
    generic_types.append(Geometry)

GenericDtype = TypeVar(
    "GenericDtype",
    *generic_types,
    covariant=True,
)
```

Reply:
Python doesn't seem to like that - complains that `A single constraint is not allowed` 🙁 

I tried
```
generic_types = [
    bool,
    int,
    str,
    float,
    pd.core.dtypes.base.ExtensionDtype,
    Bool,
    DateTime,
    Timedelta,
    Category,
    Float,
    Float16,
    Float32,
    Float64,
    Int,
    Int8,
    Int16,
    Int32,
    Int64,
    UInt8,
    UInt16,
    UInt32,
    UInt64,
    INT8,
    INT16,
    INT32,
    INT64,
    UINT8,
    UINT16,
    UINT32,
    UINT64,
    Object,
    String,
    STRING,
]

try:
    from pandas_engine import Geometry  # : ``"geometry"`` geopandas dtype

    geometry_types = [Geometry]
except ImportError:
    geometry_types = []

# expand multiple lists to get around TypeVar requiring multiple arguments for constraints
GenericDtype = TypeVar(
    "GenericDtype",
    *generic_types,
    *geometry_types,
    covariant=True,
)
```

But that doesn't work either.

Reply:
I could not find an alternative either. I think I already hit this roadblock when developing SchemaModel 🙃

Reply:
There is another related mypy error:

> pandera/typing/geopandas.py:14: error: Value of type variable "GenericDtype" of "Series" cannot be "Geometry"  [type-var]
Found 1 error in 1 file (checked 85 source files)

fixed with

```python
try:
    Geometry = pandas_engine.Geometry # TyperVar wants a variable... 
    GEOPANDAS_INSTALLED = True
except ImportError:
    GEOPANDAS_INSTALLED = False
``` 
No idea why it works though...